lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -1821,18 +1821,30 @@ static void fixAvailabilityForDecl(SourceRange ReferenceRange, const Decl *D,
   if (TypeChecker::diagnosticIfDeclCannotBePotentiallyUnavailable(D).has_value())
     return;
 
-  if (getActiveAvailableAttribute(D, Context)) {
-    // For QoI, in future should emit a fixit to update the existing attribute.
-    return;
-  }
-
   // For some declarations (variables, enum elements), the location in concrete
   // syntax to suggest the Fix-It may differ from the declaration to which
   // we attach availability attributes in the abstract syntax tree during
   // parsing.
   const Decl *ConcDecl = concreteSyntaxDeclForAvailableAttribute(D);
 
   DescriptiveDeclKind KindForDiagnostic = ConcDecl->getDescriptiveKind();
+
+  if (auto Attr = getActiveAvailableAttribute(D, Context)) {
+    if (Attr->isUnconditionallyUnavailable()) {
+      return;
+    }
+    SourceManager &SM = Context.SourceMgr;
+    auto Version = SM.extractText(Lexer::getCharSourceRangeFromSourceRange(
+                                      SM, Attr->IntroducedRange))
+                       .str();
+    auto Required = RequiredRange.getLowerEndpoint().getAsString();
+    Context.Diags
+        .diagnose(ReferenceRange.Start, diag::update_availability_attribute,
+                  KindForDiagnostic, Attr->platformString(), Version, Required)
+        .fixItReplace(Attr->IntroducedRange, Required);
```

Comment:
This looks like it'll do the right thing for attributes like:

* `@available(*, unavailable)` (just skips it)
* `@available(macOS, introduced: 10.15)`
* `@available(macOS 10.15, iOS 13, *)`

But what if you have something like these? Are there any tests cases like them?

* `@available(macOS, unavailable)`
* `@available(macOS, deprecated: 12)`

(You could just bail out for these like you do for unconditionally unavailable attributes, or you could make your fix-it insert `, introduced: 11` after the platform name in these cases.)


Also:

* `@available(*, deprecated)`

(For this, you probably either have to bail out or you have to act as though there is no active attribute and offer to insert a new one.)

Reply:
For the `deprecated`, I personally believe it is unnecessary to continue expanding to support new capabilities. 
The same for unavailable methods, unless a developer actively removes the `unavailable` attribute, we should not consider expanding to support new capabilities.   
Therefore, I think `return` is a good idea.

Reply:
Okay. If you don't want to handle those cases (except by `return`ing), I would file a follow-up issue suggesting that they could be supported in the future.

Reply:
I would also suggest add test cases for those kind of attributes in this PR as extra scenarios if we don't already.